File: 9 - PalindromeNumber/Solution2.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if (0 < x && x < 10){
        return true;
    }
    if (x < 0 || (x % 10 == 0 && x != 0)){
        return false;
    }
    let half = 0;
    while (x > half){
        half = half * 10 + x % 10;
        x = Math.floor(x /= 10);
    }

    return(x == half || x == Math.floor(half / 10));
};

File: 009 - PalindromeNumber/Solution2.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if 0 < x < 10:
            return True
        if (x < 0 or x % 10 == 0 and x != 0):
            return False
        half = 0
        while x > half:
            half = half * 10 + x % 10
            x //= 10
        return (x == half or x == half // 10)
